spec/requests/contact_spec.rb

```diff
@@ -0,0 +1,189 @@
+def compare_array_of_objects(base_array, target_array)
+  base_array.each_with_index do | item, index |
+    target = target_array[index]
+    item.each do | key, value |
+      expect(target[key]).to eq(value)
+    end
+  end
+end
+
+RSpec.describe :Contact, type: :request do
+  describe 'GET /index' do
+    let!(:contacts_list) { create_list(:contact, 2) }
+    before { get contacts_path, headers: headers }
+    
+    it 'returns 200 as http response' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'returns the correct body' do
+      parsed_contacts = parse_json
+
+      expect(parsed_contacts.size).to eq(2)
+
+      parsed_contacts.each do |contact|
+        check_attributes(contact, contacts_list.find {|c| c.id === contact[:id]} )
+      end
+    end
+  end
+
+  describe 'GET /show' do
+    before { get contact_path(target_contact) }
+
+    context 'when the contact exists' do
+      let(:target_contact) { create(:contact) }
+
+      it 'returns 200 as http response' do        
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns the correct body' do        
+        contact = parse_json
+
+        check_attributes(contact, target_contact)
+        expect(contact[:phone_numbers]).to eq([])
+      end
+    end
+
+    context 'when the contact is not found' do
+      let(:target_contact) { 'inexistent' }
+
+      it 'returns 404 as http response' do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'returns error message in the body' do
+        expect(parse_json[:message]).to eq("Couldn't find Contact with 'id'=inexistent")
+      end
+    end
+  end
+
+  describe 'POST /create' do
+    subject(:create_contact) do
+      post contacts_path, params: { contact: contact_params }
+    end
+
+    context 'with valid data' do
+      let(:contact_params) do
+        attributes_for(:contact)
+      end
+
+      it 'returns 200 as http response' do
+        create_contact
+
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns the correct body' do 
+        create_contact
+
+        contact = parse_json
+
+        check_attributes(contact, contact_params)
+        expect(contact[:phone_numbers]).to eq([])
+      end
+
+      it 'persists register in the database' do
+        expect { create_contact }.to change { Contact.count }.from(0).to(1)
+      end
+    end
+
+    context 'with valid phone_numbers' do
+      let!(:phone_number_params) do
+        { phone_numbers_attributes: [attributes_for(:phone_number), attributes_for(:phone_number)] }
+      end
+      
+      let(:contact_params) do
+        attributes_for(:contact).merge(phone_number_params)
+      end
+
+      it 'returns 200 as http response' do
+        create_contact
+
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns the correct body' do 
+        create_contact
+
+        contact = parse_json
```

Comment:
`expect(contact[:phone_numbers]).to include(phone_number_params[:phone_numbers_attributes].first)`

eu acho que isso poderia substituir o metodo `compare_array_of_objects`